### PR TITLE
testcase: add a testcase for second fetch of selectResponse

### LINKED ISSUE #37226: txn: allow larger initial chunk size for read check-ts for read-consistency
## Enhancement

When the read check-ts [feature](https://docs.pingcap.com/zh/tidb/dev/system-variables#tidb_rc_read_check_ts%E4%BB%8E-v600-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) is used, there's a constraint that the statement retry could work only when conflicts are detected and there're no result packets are responded to the client yet.

However, the init chunk size has a limit [upper bound](https://docs.pingcap.com/tidb/dev/system-variables#tidb_init_chunk_size) which is 32, which means the select statement may respond error to the client if the total result row number is larger than 32 and there's possible conflict after the first chunk is responded. As a result, the client may receive errors and have to process these errors if the read check-ts feature is enabled and the whole result row number is large, even the conflict happens rarely.

An enhancement for the read check-ts feature is to allow the init chunk size to be adjusted to a larger value like 1024 so the client does not need to handle errors when the conflict happens rarely and the select result row number is large.

### PR DESCRIPTION
**Pull request by TonsnakeLin**
Signed-off-by: TonsnakeLin <lpbgytong@163.com>

### What problem does this PR solve?

Issue Number: Ref #37226

Problem Summary:

### What is changed and how it works?
Adding a test case which tunes the system variable `tidb_max_chunk_size` to larger and gets the result from one select response.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by TonsnakeLin** [comment]
@cfzjywxk @ekexium @you06

File: server/tidb_test.go [hunk 1]
```diff
@@ -2574,6 +2578,52 @@ func TestLocalhostClientMapping(t *testing.T) {
...
+	err := cc.handleQuery(ctx, "select * from t")
+	require.NoError(t, err)
+
+	err = cc.handleQuery(ctx, "select * from t t1 join t t2")
+	require.Equal(t, kv.ErrWriteConflict, err)
+
+	tk.MustExec("set session tidb_max_chunk_size = 4096")
+	require.Equal(t, "4096", tk.MustQuery("show variables like 'tidb_max_chunk_size'").Rows()[0][1])
+	err = cc.handleQuery(ctx, "select * from t t1 join t t2")
+	require.NoError(t, err)
```
Comment (cfzjywxk) [inline server/tidb_test.go:2621]: It seems though it's not an RC transaction the statement is still retried?

Comment (TonsnakeLin) [reply server/tidb_test.go:2621]: The case can't be retried, because it makes an error at the second round of `Next`

**Review by cfzjywxk — APPROVED** [review]
Need to change `Close: #xxx` to `Ref: #xxx` in the description as there're still some other work to be done for the issue.

**Review by ekexium — COMMENTED** [review]
LGTM. Fix the lint pls.
Does this PR intend to close the original issue?

**Review by ekexium — APPROVED** [review]

**Pull request merged**